src/main/java/billgates/usecases/bill_import/BillImportUseCase.java

```diff
@@ -0,0 +1,35 @@
+package billgates.usecases.bill_import;
+
+import billgates.database.QueryEntryData;
+import billgates.entities.User;
+import billgates.interface_adapters.DatabaseGateway;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public class BillImportUseCase implements BillImportInputPort {
+    private final DatabaseGateway gateway;
+
+    public BillImportUseCase(DatabaseGateway gateway) {
+        this.gateway = gateway;
+    }
+
+    @Override
+    public void importBill(BillImportRequestModel model) {
+        List<QueryEntryData> result = this.gateway.getBillData(User.getInstance().getCurrentBillID()).getEntries();
+
+        int entryID = result.size() + 1;
```

Comment:
Actually, for the updated method of insertEntry in the database gateway, we don't need the ID anymore. 

So, you can simply instantiate a QueryEntryData without an ID, and it will still work. Please check out #31 for more details.